spec/project_spec.rb

```diff
@@ -117,4 +117,61 @@
       end
     end
   end
+
+  describe "Module functions" do
+    describe "FastlaneCore.xcode_list_timeout" do
+      before do
+        ENV['FASTLANE_XCODE_LIST_TIMEOUT'] = nil
+      end
+      it "returns default value" do
+        expect(FastlaneCore.xcode_list_timeout).to eq(10)
+      end
+      it "returns specified value" do
+        ENV['FASTLANE_XCODE_LIST_TIMEOUT'] = '5'
+        expect(FastlaneCore.xcode_list_timeout).to eq(5)
+      end
+      it "returns 0 if empty" do
+        ENV['FASTLANE_XCODE_LIST_TIMEOUT'] = ''
+        expect(FastlaneCore.xcode_list_timeout).to eq(0)
+      end
+      it "returns 0 if garbage" do
+        ENV['FASTLANE_XCODE_LIST_TIMEOUT'] = 'hiho'
+        expect(FastlaneCore.xcode_list_timeout).to eq(0)
+      end
+    end
+
+    describe "FastlaneCore.run_command" do
+      def count_processes(text)
+        `ps -aef | grep #{text} | grep -v grep | wc -l`.to_i
+      end
+
+      it "runs simple commands" do
+        cmd = 'echo "HO"'
+        expect(FastlaneCore.run_command(cmd)).to eq("HO\n")
+      end
+
+      it "runs more complicated commands" do
+        cmd = "ruby -e 'sleep 0.1; puts \"HI\"'"
+        expect(FastlaneCore.run_command(cmd)).to eq("HI\n")
+      end
+
+      it "should timeouts and kills" do
+        text = "FOOBAR" # random text
+        count = count_processes(text)
+        cmd = "ruby -e 'sleep 3; puts \"#{text}\"'"
+        # this doesn't work ?!
```

Comment:
I've seen that issue multiple times in the past weeks. E.g. https://github.com/lacostej/spaceship/blob/client_hardening/spec/client_spec.rb#L41-L43

Not sure why this is like that... Do you have any idea ?